rir/src/compiler/pir/instruction.h

```diff
@@ -112,6 +111,11 @@ enum class Effect : uint8_t {
     FIRST = Visibility,
     LAST = ExecuteCode,
 };
+// Effects an instruction causes itself. These effects can lead to
+// lower effects, e.g. if an instruction executes more code, that can cause
```

Comment:
> These effects can lead to lower effects

no, this I would like to absolutely avoid! if an instruction has effect `executeCode` but not `warning`, this means, we know that the code it executes does not print any warning. I mean we anyway don't do such fancy things. But I would like to stress that we should always set all the effect flags that an instruction can have and never assume that one of them subsumes the other.

My suggestion for grouping effects while printing, was merely since some effects are just not as important if others are present. E.g. if we have Error, then we don't care that much if it can also warning. But nevertheless Error|Warning means it can either show a warning or throw an error (or neither), whereas Error means, just error, no warning.